### PR TITLE
Bug in Series constructor returning wrong missing values

### LINKED ISSUE #43017: BUG: Default value for `Series(dtype=int)` is not `pd.NA`
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of pandas.

---

**Note**: Please read [this guide](https://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports) detailing how to provide the necessary information for us to reproduce your bug.

#### Code Sample, a copy-pastable example

```python
pd.Series(index=[0], dtype=int)

```

#### Problem description

For other types, the default missing value indicator is set properly:
```Python
pd.Series(index=[0], dtype=float)
```
0   NaN
dtype: float64

```Python
pd.Series(index=[0], dtype="datetime64[ns]")
```
0   NaT
dtype: datetime64[ns]

```Python
pd.Series(index=[0], dtype=bool)
```
0    False
dtype: bool
(This might be tricky, but understandable)

However, for integer types, the default missing value is set to 0 instead of `pd.NA`, which is dangerous for interpretation.

#### Expected Output

Either

```
0    <NA>
dtype: Int64
```

or 
```
0    NaN
dtype: float64
```

makes sense, given how `int64` is converted when having missing values, the latter case may be better.

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : c7f7443c1bad8262358114d5e88cd9c8a308e8aa
python           : 3.7.11.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
Version          : 10.0.19041
machine          : AMD64
processor        : Intel64 Family 6 Model 142 Stepping 10, GenuineIntel
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : None.None

pandas           : 1.3.1
numpy            : 1.20.3
pytz             : 2021.1
dateutil         : 2.8.2
pip              : 21.2.2
setuptools       : 52.0.0.post20210125
Cython           : 0.29.24
pytest           : 6.1.2
hypothesis       : 6.14.1
sphinx           : 4.0.2
blosc            : None
feather          : None
xlsxwriter       : 1.4.4
lxml.etree       : 4.6.3
html5lib         : 1.1
pymysql          : 1.0.2
psycopg2         : None
jinja2           : 3.0.1
IPython          : 7.22.0
pandas_datareader: None
bs4              : 4.9.3
bottleneck       : 1.3.2
fsspec           : 2021.07.0
fastparquet      : None
gcsfs            : None
matplotlib       : 3.4.2
numexpr          : 2.7.3
odfpy            : None
openpyxl         : 3.0.7
pandas_gbq       : None
pyarrow          : 3.0.0
pyxlsb           : None
s3fs             : 0.4.2
scipy            : 1.6.2
sqlalchemy       : 1.4.22
tables           : 3.6.1
tabulate         : 0.8.9
xarray           : 0.19.0
xlrd             : 2.0.1
xlwt             : 1.3.0
numba            : 0.53.0

</details>

### PR DESCRIPTION
**Pull request by phofl**
- [x] closes #43017
- [x] xref #43018 (if the reindex case is ok, then this is done too)
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [x] whatsnew entry

This ensures consistency with the DataFrame constructor

### CODE REVIEW

File: doc/source/whatsnew/v1.4.0.rst [hunk 1]
```diff
@@ -264,6 +264,7 @@ Numeric
 Conversion
 ^^^^^^^^^^
 - Bug in :class:`UInt64Index` constructor when passing a list containing both positive integers small enough to cast to int64 and integers too large too hold in int64 (:issue:`42201`)
+- Bug in :class:`Series` constructor returning 0 for missing values with dtype ``int64`` and ``False`` for dtype ``boo`` (:issue:`43017`, :issue:`43018`)
```
Comment (jreback) [inline doc/source/whatsnew/v1.4.0.rst:267]:
```suggestion
- Bug in :class:`Series` constructor returning 0 for missing values with dtype ``int64`` and ``False`` for dtype ``bool`` (:issue:`43017`, :issue:`43018`)
```

File: pandas/tests/series/methods/test_astype.py [hunk 2]
```diff
@@ -356,6 +356,13 @@ def test_astype_bytes(self):
         result = Series(["foo", "bar", "baz"]).astype(bytes)
         assert result.dtypes == np.dtype("S3")
 
+    def test_astype_nan_to_bool(self):
+        # GH#43018
+        ser = Series(np.nan, dtype="object")
+        result = ser.astype("bool")
+        expected = Series(True, dtype="bool")
```
Comment (jreback) [inline pandas/tests/series/methods/test_astype.py:363]: this is True?

File: pandas/tests/series/test_constructors.py [hunk 3]
```diff
@@ -1776,6 +1776,18 @@ def test_constructor_with_pandas_dtype(self):
...
+    def test_constructor_int_dtype_missing_values(self):
+        # GH#43017
+        result = Series(index=[0], dtype="int64")
+        expected = Series(np.nan, index=[0], dtype="float64")
+        tm.assert_series_equal(result, expected)
+
+    def test_constructor_bool_dtype_missing_values(self):
+        # GH#43018
+        result = Series(index=[0], dtype="bool")
+        expected = Series(True, index=[0], dtype="bool")
```
Comment (jreback) [inline pandas/tests/series/test_constructors.py:1788]: this is True?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/series/methods/test_astype.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/tests/series/methods/test_astype.py:363]: `bool(np.nan)` returns True

Comment (phofl) [reply pandas/tests/series/methods/test_astype.py:363]: Yes exactly.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @phofl

**Comment by simonjayhawkins** [comment]
@phofl we have had reports regarding both these cases. Is the new behavior now consistent?

```python
import pandas as pd

print(pd.__version__)
s = pd.Series(dtype="int", index=[0])
print(s)
s2 = pd.Series(dtype="bool", index=[0])
print(s2)
```
```
1.3.5
0    0
dtype: int64
0    False
dtype: bool
```
```
1.4.1
0   NaN
dtype: float64
0    True
dtype: bool
```

It appears to me (and maybe others from the issues) that the changes are confusing.

We have changed the `int` case and and instead create a `float` array saying that the missing value cannot be held in a integer array and yet for the bool case we continue to keep the bool dtype even though we cannot represent a missing value in a boolean array?

I appreciate that `int(np.nan)` raises `ValueError: cannot convert float NaN to integer` and `bool(np.nan)` is `True` but the users are not specifying `np.nan`, they are not supplying data. I wonder whether we ought to revert this until we change the constructors to use nullable dtypes?

**Comment by jbrockmendel** [comment]
>  I wonder whether we ought to revert this until we change the constructors to use nullable dtypes?

no opinion on the reversion, but i advise against the implicit assumption that constructors are going to default to nullable dtypes.  support has improved, but its still a mess.

**Pull request merged**